migration/migration.go

```diff
@@ -163,3 +175,93 @@ func recalculateBloomFilters(txn db.Transaction) error {
 		}
 	}
 }
+
+var trieNodeBuckets = map[db.Bucket]struct {
+	seekTo  []byte
+	skipLen int
+}{
+	db.ClassesTrie: {
+		seekTo:  db.ClassesTrie.Key(),
+		skipLen: 1,
+	},
+	db.StateTrie: {
+		seekTo:  db.StateTrie.Key(),
+		skipLen: 1,
+	},
+	db.ContractStorage: {
+		seekTo:  db.ContractStorage.Key(),
+		skipLen: 1 + felt.Bytes,
+	},
+}
+
+type trieNode struct {
+	Value *felt.Felt
+	Left  *bitset.BitSet
+	Right *bitset.BitSet
+}
+
+func changeTrieNodeEncoding(txn db.Transaction) error {
+	var n trieNode
+	var buf bytes.Buffer
+	var updatedNodes uint64
+
+	migrateF := func(it db.Iterator, bucket db.Bucket, seekTo []byte, skipLen int) error {
+		bucketPrefix := bucket.Key()
+		for it.Seek(seekTo); it.Valid(); it.Next() {
+			key := it.Key()
+			if !bytes.HasPrefix(key, bucketPrefix) {
+				delete(trieNodeBuckets, bucket)
+				break
+			}
+
+			if len(key) == skipLen {
+				// this entry is not a TrieNode, it is the root key
+				continue
+			}
+
+			const updatedNodesBatch = 1_000_000
```

Comment:
We need this because a single transaction can't modify more than 1M entries, right? Can we document this behavior?

Reply:
> a single transaction can't modify more than 1M entries, right

We cant fit the entire migration in a single transaction but the actual limit is much higher than 1M. But the more updates you queue on a transaction the more memory it uses. 1M is just an arbitrary number that we can both fit in a transaction and dont use too much memory.

Copying this verbatim to source code:)

Reply:
Got it, makes much more sense. Thanks!